tests/test_ci_unit_test_output_guard.sh

```
@@ -0,0 +1,83 @@
+
+if [ ! -x "$SCRIPT" ]; then
+  echo "FAIL: missing executable hard-failure guard at $SCRIPT" >&2
+  exit 1
+fi
+
+if ! grep -Fq './tests/test_ci_unit_test_output_guard.sh' "$WORKFLOW_FILE"; then
+  echo "FAIL: ci.yml must run the unit-test hard-failure guard test" >&2
+  exit 1
+fi
+
+if ! grep -Fq './scripts/ci-unit-test-output-guard.sh "$EXIT_CODE" /tmp/test-output.txt' "$WORKFLOW_FILE"; then
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)